master/internal/provisioner/scale_decider.go

```diff
@@ -110,15 +124,47 @@ func (s *scaleDecider) findInstancesToTerminate(
 			toTerminate[inst.ID] = true
 
 		case Running:
-			if _, ok := s.agentSnapshot[inst.AgentName]; ok {
+			if _, ok := s.idleAgentSnapshot[inst.AgentName]; ok {
 				idleInstances[inst.ID] = true
 			}
 		}
 	}
 
+	// Identify instances that are not currently connected to the master.
+	now := time.Now()
+	for _, inst := range s.instanceSnapshot {
```

Comment:
non-blocking: if you addressed the two comments below in this file, It would be just easy to merge this block into the above block.

Reply:
not sure what's up with GH bu I only see one other comment and it doesn't seem super related to this, but I did merge this into the earlier block.